test/components/popup.spec.tsx

```diff
@@ -28,68 +50,82 @@ describe('<Popup />', function() {
 
             await waitForDom(() => {
-                expect(select(demoContainer)).to.be.present();
-                expect(select(demoContainer, portalId)).to.be.absent();
+                expect(popupDemo.container).to.be.present();
+                expect(popupDemo.isPopupPresent).to.be.false;
             });
-            select<HTMLDivElement>(demoContainer)!.click();
-            await waitForDom(() => expect(bodySelect(portalId)).to.be.present());
-            select<HTMLDivElement>(demoContainer)!.click();
-            return waitForDom(() => expect(bodySelect(portalId)).to.be.absent());
+
+            (popupDemo.container as HTMLDivElement).click();
+            popupDemo = new PopupDemoTestDriver(result as PopupDemo);
+            await waitForDom(() => expect(popupDemo.portal).to.be.present());
+
+            (popupDemo.container as HTMLDivElement).click();
+            popupDemo = new PopupDemoTestDriver(result as PopupDemo);
+            return waitForDom(() => expect(popupDemo.isPopupPresent).to.be.false);
         });
     });
 
     it('displays the popup and renders its children if the open prop is given', function() {
-        clientRenderer.render(
+        const {result} = clientRenderer.render(
             <Popup anchor={anchor} open>
-                <span data-automation-id="SPAN">Popup Body</span>
+                <span>Popup Body</span>
             </Popup>
         );
 
+        const popup = new PopupTestDriver(result as Popup);
+
         return waitFor(() => {
-            expect(bodySelect(portalId)).to.be.present();
-            expect(bodySelect(portalId, 'SPAN')).to.be.present();
+            expect(popup.root).to.be.present();
+            expect(popup.content[0]).to.be.present();
         });
     });
 
     it('does not render the popup if there is no anchor', async function() {
-        clientRenderer.render(
+        const {result} = clientRenderer.render(
             <Popup anchor={null} open={true}>
-                <span data-automation-id="SPAN">Popup Body</span>
+                <span>Popup Body</span>
             </Popup>
         );
         await sleep(100);
-        await waitFor(() => expect(bodySelect(portalId)).to.be.absent());
+
+        const popup = new PopupTestDriver(result as Popup);
+        await waitFor(() => expect(popup.isPresent).to.be.false);
     });
 
     it('does not reder the popup if the open prop is false', async function() {
-        clientRenderer.render(
+        const {result} = clientRenderer.render(
             <Popup anchor={anchor} open={false}>
                 <span data-automation-id="SPAN">Popup Body</span>
             </Popup>
         );
         await sleep(100);
-        await waitFor(() => expect(bodySelect(portalId)).to.be.absent());
+
+        const popup = new PopupTestDriver(result as Popup);
+        await waitFor(() => expect(popup.isPresent).to.be.false);
     });
 
     it('removes the component when unmounting', async function() {
```

Comment:
This test is not working as intended (Problem is with the original test) and it is possibly not required since Portal tests for the same thing.
If we choose to keep this test, another bug was added to it, so we'll discuss it if needed